### ISSUE TITLE
Per function role without (or with empty) iamRoleStatements

### ISSUE THREAD

**Issue by jogold** [functionalone/serverless-iam-roles-per-function#9]
Hello,

Thank you for this great plugin!

I have the following `serverless.yml`:

```
functions:
  firstFunction:
    handler: functions/firstFunction.handler

  secondFunction:
    handler: functions/secondFunction.handler
    iamRoleStatements:
      - Effect: Allow
        Action: sns:Publish
        Resource: '*'

  thirdFunction:
    handler: functions/thirdFunction.handler
    vpc:
      securityGroupIds:
        - sg-xxxxxx
      subnetIds:
        - subnet-xxxx
        - subnet-xxxxx
```

`firstFunction` and `thirdFunction` get the global role, so they both have VPC policy and permission to write in all logs.

I tried adding `iamRoleStatements: []` but it doesn't work due to the early return here https://github.com/functionalone/serverless-iam-roles-per-function/blob/master/src/lib/index.ts#L149.

Any specific reason for this early return in `createRoleForFunction`? I took a quick look at the code and it seems that this can be safely removed?

**Comment by jogold** [comment]
Hi @glicht,

Yes, the issue is with the first and third function not receiving a per function role and the fact that even when using `iamRoleStatements: []` it doesn't work.

**Comment by glicht** [comment]
Ok. Thanks for the clarification. I think this can be fixed easily. Will check this out soon.

**Comment by buggy** [comment]
I think something like this should do the trick

```yaml
custom:
  defaultIamRoleStatements: &defaultIamRoleStatements
    # Put your default IAM statements here

functions:
  firstFunction:
    handler: functions/firstFunction.handler
    iamRoleStatements:
      <<: *defaultIamRoleStatements

  secondFunction:
    handler: functions/secondFunction.handler
    iamRoleStatements:
      - Effect: Allow
        Action: sns:Publish
        Resource: '*'

  thirdFunction:
    handler: functions/thirdFunction.handler
    iamRoleStatements:
      <<: *defaultIamRoleStatements
    vpc:
      securityGroupIds:
        - sg-xxxxxx
      subnetIds:
        - subnet-xxxx
        - subnet-xxxxx
```

**Comment by glicht** [comment]
@buggy thanks for the workaround suggestion. Anyway, I just published a new version (0.1.9) which should fix this issue.

**Comment by jogold** [comment]
Thanks @glicht!
